### ISSUE TITLE
linter: warn unmatched rule names

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#6763]
Given 

```
  "rules": {
    "for-direction": "error",
    "no-async-promise-executor": "error",
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
...
```

I don't know which rule is applied or not, we should warn the ones that are incorrect.
